tests/conftest.py

```diff
@@ -47,9 +47,23 @@ def restore_env_variables():
     """Ensures that environment variables set during the test do not leak out."""
     env_backup = os.environ.copy()
     yield
+    leaked_vars = os.environ.keys() - env_backup.keys()
     # restore environment as it was before running the test
     os.environ.clear()
     os.environ.update(env_backup)
+    # these are currently known leakers - ideally these would not be allowed
```

Comment:
I guess some of these would always stay:

- "CUDA_DEVICE_ORDER": not set by us
- "PL_GLOBAL_SEED", "PL_SEED_WORKERS": we want the same values across jobs